pydantic/networks.py

```diff
@@ -329,6 +329,12 @@ def __init__(self, name: str, email: str):
         self.name = name
         self.email = email
 
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, NameEmail):
+            return (self.name, self.email,) == (other.name, other.email,)
+        else:
+            return other == self
```

Comment:
```suggestion
        return isinstance(other, NameEmail) and (self.name, self.email) == (other.name, other.email)
```
?

Reply:
Or should this also compare to the string representation?
```py
return (isinstance(other, NameEmail) and (self.name, self.email) == (other.name, other.email)) or str(self) == other
```
?

Reply:
surely too much magic?

Reply:
Yeah, not a fan of that magic. Changes applied.